### PR TITLE
Fix module loading

### PR DESCRIPTION
**Pull request by mailwl**
Calling .init_proc() from loaded modules and stub libkernel functions needed for this
Relocate GLOB_DAT

### CODE REVIEW

File: GPCS4/SceModules/SceLibkernel/sce_kernel_types.h [hunk 1]
```diff
@@ -6,6 +6,35 @@ typedef struct _SceKernelSemaOptParam {
...
+{
+	uint64_t size{ 0x40 };
+	uint64_t p1{};
+	uint64_t p2{};
+	uint64_t p3{};
+	uint64_t p4{};
+	uint64_t p5{};
+	uint64_t p6{};
+	SubProcParam* subProcParam{};
+} ProcParam;
```
Comment (VelocityRa) [inline GPCS4/SceModules/SceLibkernel/sce_kernel_types.h]: not necessary, see other comment

File: GPCS4/SceModules/SceLibkernel/sce_libkernel.cpp [hunk 2]
```diff
@@ -141,6 +146,64 @@ int PS4API sceKernelUuidCreate(void)
 }
 
 
+ProcParam* PS4API sceKernelGetProcParam(uint64_t p1, uint64_t p2)
+{
+	LOG_SCE_DUMMY_IMPL();
+	static SubProcParam subProcParam;
+	static ProcParam procParam;
+	procParam.subProcParam = &subProcParam;
+	return &procParam;
```
Comment (VelocityRa) [inline GPCS4/SceModules/SceLibkernel/sce_libkernel.cpp]: This data is inside the binary in the `PT_SCE_PROCPARAM` segment.

File: GPCS4/Loader/ELFMapper.cpp [hunk 3]
```diff
@@ -261,7 +261,7 @@ bool ELFMapper::mapImageIntoMemroy()
 			if (phdr.p_flags & PF_X)
 			{
 				retVal = mapCodeSegment(phdr);
-				LOG_DEBUG("code segment at 0x%08x size=%ld", info.pCodeAddr,
+				LOG_DEBUG("code segment at 0x%x size=%ld", info.pCodeAddr,
```
Comment (DummyCatz) [inline GPCS4/Loader/ELFMapper.cpp]: Maybe we should not use %x here because pointers and unsigned ints are not the same size in x64. What about %p?

Comment (mailwl) [reply GPCS4/Loader/ELFMapper.cpp]: %p shows a different way on differ platforms. For example, on macOS it shows as "0x....", on Windows without "0x"

File: GPCS4/SceModules/SceLibkernel/sce_libkernel.cpp [hunk 4]
```diff
@@ -141,6 +146,84 @@ int PS4API sceKernelUuidCreate(void)
...
+{
+	LOG_FIXME("Not implemented");
+	return SCE_OK;
+}
+
+
+void PS4NORETURN PS4API sceKernelDebugRaiseException(uint32_t error_code, uint32_t param)
+{
+	LOG_SCE_DUMMY_IMPL();
+	exit(-1);
```
Comment (VelocityRa) [inline GPCS4/SceModules/SceLibkernel/sce_libkernel.cpp]: Do a `trapDebugger` before exiting maybe.
Same for libc `exit`.

**Comment by Inori** [comment]
e8d9e1af53accdc4c3d8ee9bfd2a411edae4f6fd

**Pull request merged**